hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/java/org/apache/hadoop/yarn/nodelabels/store/AbstractFSNodeStore.java

```diff
@@ -65,8 +65,30 @@ protected void initStore(Configuration conf, Path fsStorePath,
     this.fsWorkingPath = fsStorePath;
     this.manager = mgr;
     initFileSystem(conf);
-    // mkdir of root dir path
-    fs.mkdirs(fsWorkingPath);
+    // mkdir of root dir path with retry logic
+    int maxRetries = 3;
```

Comment:
Can you please create a configuration entry for the retry count and the interval? Something like the 
yarn.resourcemanager.zk-num-retries and yarn.resourcemanager.connect.retry-interval.ms parameters defined for ZK connections. That way the user could configure a longer period in case HDFS in safe mode for a few minutes when the RM tries to start.

Reply:
Thanks @brumi1024 for your review. I hve made this change in last commit